File: Introduction to Data Structures: Hash Table/Practical Application - Hash Map/Two Sum/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map <Integer, Integer> hashMap = new HashMap<>(); 
        int [] result = new int [2];
        int diff;
        
        for (int i = 0; i < nums.length; i++) {
            
            diff = target - nums[i];
            
            if (hashMap.containsKey(diff)) {
                result[0] = i;
                result[1] = hashMap.get(diff);
                return result;
            }
            else {
                hashMap.put(nums[i], i);
            }
        }
        return result;
    }
}

File: Introduction to Data Structures: Hash Table/3. Practical Application - Hash Map/Two Sum/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        hashMap = {}  # also called dictionary in Python

        for i in range(0, len(nums)):

            diff = target - nums[i]

            if diff in hashMap.keys():

                return [i, hashMap[diff]]

            else:

                hashMap[nums[i]] = i
